demos/noise_suppression_demo/cpp/main.cpp

```diff
@@ -116,131 +117,152 @@ void read_wav(const std::string& file_name, std::vector<int16_t>& wave, RiffWave
         }
 
+        slog::info << ov::get_openvino_version() << slog::endl;
+
         // Loading Inference Engine
-        InferenceEngine::Core ie;
+        ov::runtime::Core core;
 
-        InferenceEngine::CNNNetwork network = ie.ReadNetwork(FLAGS_m);
-        InferenceEngine::InputsDataMap inputs = network.getInputsInfo();
+        std::shared_ptr<ov::Model> model = core.read_model(FLAGS_m);
+        slog::info << "model file: " << FLAGS_m << slog::endl;
+        logBasicModelInfo(model);
 
-        //get state names pairs (inp,out)
-        std::vector<std::pair<std::string, std::string>> state_names;
+        ov::OutputVector inputs = model->inputs();
+        ov::OutputVector outputs = model->outputs();
+
+        // get state names pairs (inp,out) and compute overall states size
         size_t state_size = 0;
-        for(auto& inp: inputs) {
-            std::string inp_state_name = inp.first;
-            if (inp_state_name.find("inp_state") == std::string::npos)
+        std::vector<std::pair<std::string, std::string>> state_names;
+        for (size_t i = 0; i < inputs.size(); i++) {
+            std::string inp_state_name = inputs[i].get_any_name();
+            if (inp_state_name.find("inp_state_") == std::string::npos)
                 continue;
+
             std::string out_state_name(inp_state_name);
             out_state_name.replace(0, 3, "out");
+
+            // find corresponding output state
+            auto scmp = [&](ov::Output<ov::Node> output) { return output.get_any_name() == out_state_name; };
+            if (std::end(outputs) == std::find_if(outputs.begin(), outputs.end(), scmp))
+                throw std::logic_error("model output state name not corresponf input state name");
+
             state_names.emplace_back(inp_state_name, out_state_name);
-            const InferenceEngine::SizeVector& size = inputs[inp_state_name]->getInputData()->getTensorDesc().getDims();
+
+            ov::Shape shape = inputs[i].get_shape();
             size_t tensor_size = 1;
-            for(size_t s: size) {
+            for (size_t s : shape)
                 tensor_size *= s;
-            }
+
             state_size += tensor_size;
         }
-        std::cout << "State_param_num = " << state_size << " (" << state_size*4e-6 << "Mb)" << std::endl;
-        InferenceEngine::ExecutableNetwork executable_network = ie.LoadNetwork(network, FLAGS_d);
-        InferenceEngine::InferRequest infer_request = executable_network.CreateInferRequest();
+
+        if (state_size == 0)
+            throw std::logic_error("no expected model state inputs found");
+
+        std::cout << "State_param_num = " << state_size << " (" << state_size * 4e-6f << "Mb)" << std::endl;
+
+        // sort names
+        auto scmp = [](const std::pair<std::string, std::string>& a, const std::pair<std::string,std::string>& b) { return a.first < b.first; };
+        std::sort(state_names.begin(), state_names.end(), scmp);
```

Comment:
Does the model will work correctly without sorting the names? I suppose so. After the inference the `inp_state_*` is filled by the specific `out_state_*` and the order is does not matter, because the sequential number of `inp_state_*` is compared with its shape (the states have different shapes)

Reply:
did not get what you are saying

Reply:
that's correct, input state should correspond to output state, but the order in which output states from previous inference copied to input state for the next  inference does not matter while all states copied, so sorting removed